model/task/local_service.go

```diff
@@ -116,19 +143,86 @@ func (s *localService) Get(ctx context.Context, taskOpts []Task, fields ...strin
 	}
 
 	var allDBTaskResults []testresult.DbTaskTestResults
-	cur, err := s.env.DB().Collection(testresult.Collection).Find(ctx, filter, opts)
+	cur, err := s.env.CedarDB().Collection(testresult.Collection).Find(ctx, filter, opts)
 	if err != nil {
 		return nil, errors.Wrap(err, "finding DB test results")
 	}
 	if err = cur.All(ctx, &allDBTaskResults); err != nil {
 		return nil, errors.Wrap(err, "reading DB test results")
 	}
-
 	allTaskResults := make([]testresult.TaskTestResults, len(allDBTaskResults))
+
+	if len(fields) == 0 {
+		toDownload := make(chan *testresult.DbTaskTestResults, len(allDBTaskResults))
+		for i := range allDBTaskResults {
+			toDownload <- &allDBTaskResults[i]
+		}
+		close(toDownload)
+
+		config, err := evergreen.GetConfig(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "retrieving config")
+		}
+
+		var wg sync.WaitGroup
+		catcher := grip.NewBasicCatcher()
+		for i := 0; i < runtime.NumCPU(); i++ {
+			wg.Add(1)
+			go func() {
```

Comment:
nit: since this function is quite complex and long, could we move the goroutine worker logic into a helper function?